File: Python/programmers/etc/level1/명예의_전당1.py
from heapq import *

def solution(k, score):
    answer = []
    q = []
    for s in score:
        heappush(q, s)
        if len(q) == k+1:
            heappop(q)
        answer.append(q[0])
    return answer


k = 3
score = [10, 100, 20, 150, 1, 100, 200]
print(solution(k, score)) # [10, 10, 10, 20, 20, 100, 100]

k = 4
score = [0, 300, 40, 300, 20, 70, 150, 50, 500, 1000]
print(solution(k, score)) # [0, 0, 0, 0, 20, 40, 70, 70, 150, 300]

File: Java/programmers/etc/level1/명예의전당1.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 명예의전당1 {
        public List<Integer> solution(int k, int[] score) {
        List<Integer> answer = new ArrayList<>();
        PriorityQueue<Integer> q = new PriorityQueue<>();
        
        for (int s : score) {
            q.add(s);
            if (q.size() == k+1) {
                q.poll();
            }
            answer.add(q.peek());
        }
        
        return answer;
    }
}
